pkg/engine/task/task_apply.go

```diff
@@ -139,3 +182,37 @@ func isHealthy(ro []runtime.Object) error {
 	}
 	return nil
 }
+
+// namespacedGVKResources method uses the discovery client to fetch all API resources (with Groups and Versions)
+// and returns a mapping of GVK -> isNamespaced. Useful when deciding whether a given object is a cluster-scoped resource.
+func namespacedGVKResources() (map[schema.GroupVersionKind]bool, error) {
+	namespaced := map[schema.GroupVersionKind]bool{}
+
+	restCfg, err := config.GetConfig()
```

Comment:
you should probably inject the config rather than load it here. You would have to find a good place from which to pass it in though, the most upper level for this is `manager.GetConfig` in `main.go` and then carry it through controller